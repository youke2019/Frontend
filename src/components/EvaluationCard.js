import React from "react";
import {
    Text,
    StyleSheet,
    View,
    Image,
    TouchableOpacity,
} from "react-native";
import {Avatar} from "react-native-elements"
import Rating from './Rating'
import {UserIdText} from '../components/UserIdText'

class EvaluationCard extends React.Component {
    state = {
        showDetail: false,
        detail: [],
    }

    componentWillMount() {
        let detail = []
        for (let item in this.props.evaluation.evaluate_content)
            if (item != 'course_id' && item != 'user_id' && item != 'credit_point' && item != '课程简述' && item != 'evaluate_id')
                detail.push({
                    title: item,
                    content: this.props.evaluation.evaluate_content[item]
                })
        this.setState({
            detail: detail
        })
    }

    switchDetailState = () => {
        this.props.onDetail()
        this.setState({
            showDetail: !this.state.showDetail
        })
    }

    render(){
        const {
            evaluation={}
        } = this.props

        const {
            detail,
            showDetail,
        } = this.state

        return (
            <View
                style={styles.background}
            >
                <View style={styles.container}>
                    <Image
                        style={styles.header}
                        source={{uri:'uber'}}
                        resizeMode='cover'
                    >
                    </Image>
                    <View style={styles.outline}>
                        <Text>授课教师: {evaluation.evaluate_content.授课教师}</Text>
                    </View>
                    <View style={styles.star}>
                        <View style={styles.star_icon_container}>
                            <Text style={styles.star_text}>评分:</Text>
                            <Rating
                                rate={evaluation.evaluate_content.credit_point}
                            />
                        </View>
                        <Text style={styles.star_number}>{evaluation.evaluate_content.credit_point}</Text>
                    </View>
                    {
                        showDetail &&
                        <View style={styles.details}>
                            {
                                detail.map((item,index) =>
                                    <View
                                        style={styles.detail}
                                        key={index}
                                    >
                                        <View style={styles.detail_title_container}>
                                            <Text style={styles.detail_title}>{item.title}</Text>
                                        </View>
                                        <View style={styles.detail_content_container}>
                                            <Text style={styles.detail_content}>{item.content}</Text>
                                        </View>
                                    </View>
                                )
                            }
                        </View>
                    }
                    <View style={styles.bottom}>
                        <View style={styles.user}>
                            <Avatar
                                size="small"
                                rounded
                                source={{uri: 'https://s3.amazonaws.com/uifaces/faces/twitter/adhamdannaway/128.jpg'}}
                            />
                            <UserIdText
                                style={styles.user_name}
                                user_id={evaluation.user_id}
                            />
                        </View>
                        <TouchableOpacity
                            onPress={this.switchDetailState}
                            style={styles.showDetail}
                        >
                            <Text>查看详情</Text>
                            <Image
                                style={styles.arrow}
                                source={{uri: showDetail? 'arrow_up':'arrow_down'}}
                            />
                        </TouchableOpacity>
                    </View>
                </View>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    background:{
        backgroundColor: '#FFFFFF',
        flex:1,
        width: 300,
        borderRadius: 20,
        elevation:1,
    },
    container: {
        padding: 20,
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'stretch',
    },
    header:{
        height: 180,
        justifyContent: 'center',
        alignItems: 'center',
    },
    outline:{
        height: 80,
        paddingHorizontal: 10,
        paddingTop: 15,
        justifyContent: 'center',
        alignItems: 'center',
    },
    star:{
        height: 60,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
    },
    star_icon_container:{
        flex:7,
        flexDirection: 'row',
        alignItems: 'center',
    },
    star_text:{
        paddingRight: 10,
    },
    star_number:{
        flex:2,
        textAlign: 'right',
        fontSize: 21,
    },
    details:{
    },
    detail:{
        padding: 15,
        flex:1,
    },
    detail_title_container:{},
    detail_title:{
        fontSize: 15,
        fontWeight: '500'
    },
    detail_content_container:{
        padding: 8,
    },
    detail_content:{
    },
    bottom:{
        height: 30,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
    },
    user:{
        flexDirection: 'row',
        alignItems: 'center',
    },
    user_name:{
        paddingLeft: 5,
    },
    showDetail:{
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
    },
    arrow:{
        width: 15,
        height: 15,
    }
})

export default EvaluationCard