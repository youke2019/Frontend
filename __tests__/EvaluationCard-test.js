import React from 'react'
import renderer from 'react-test-renderer'
import {shallow, mount, render, configure} from 'enzyme'
import EvaluationCard from '../src/components/EvaluationCard'
import Adapter from "enzyme-adapter-react-16/build"

configure({adapter: new Adapter()})

const evaluation = {
    "evaluate_id": 5,
    "evaluate_time": "2019-07-22 14:21:34",
    "user_id": "1A7A1F80-59D0-4E8C-B4C6-AAAFFC606120",
    "course_id": "LI901",
    "evaluate_content": {
        "course_id": "LI901",
        "课程简述": "又水分又高",
        "credit_point": 10,
        "上课自由程度": "基本不管 坐后排随便你干嘛",
        "课程个人体验": "课是挺水的 但是没什么好玩的地方",
        "evaluate_id": 5,
        "user_id": "1A7A1F80-59D0-4E8C-B4C6-AAAFFC606120",
        "考核形式": "5分钟个人pre+1000字论文"
    },
    "evaluate_praise_point": 0,
    "current_user_praise": false,
    "courseEvaluationPraiseList": []
}

const onDetail = jest.fn(() => {})
jest.mock('axios')
global.baseUrl = 'baseUrl'

describe('functions', () => {
    let wrapper = shallow(
        <EvaluationCard
            evaluation={evaluation}
            onDetail={onDetail}
        />)

    it('switchDetailState()', () => {
        wrapper.setState({
            showDetail: false
        })
        expect(wrapper.instance().state.showDetail).toBe(false)
        wrapper.instance().switchDetailState()
        expect(wrapper.instance().state.showDetail).toBe(true)
        expect(onDetail).toHaveBeenCalled()
    })
})

test('renders QuestionCard page correctly', () => {
    const tree = renderer.create(
        <EvaluationCard
            evaluation={evaluation}
            onDetail={onDetail}
        />).toJSON()
    expect(tree).toMatchSnapshot();
})
