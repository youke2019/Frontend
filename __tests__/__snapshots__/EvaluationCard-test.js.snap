// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EvaluationCard page correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 20,
      "flex": 1,
      "width": 300,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": "uber",
        }
      }
      style={
        Object {
          "alignItems": "center",
          "height": 180,
          "justifyContent": "center",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 80,
          "justifyContent": "center",
          "paddingHorizontal": 10,
          "paddingTop": 15,
        }
      }
    >
      <Text>
        又水分又高
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 7,
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "paddingRight": 10,
            }
          }
        >
          评分:
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "paddingHorizontal": 5,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "height": 25,
                  "width": 25,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "fullstar1",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": 25,
                      "width": 25,
                    },
                    Object {
                      "resizeMode": "cover",
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "paddingHorizontal": 5,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "height": 25,
                  "width": 25,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "fullstar2",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": 25,
                      "width": 25,
                    },
                    Object {
                      "resizeMode": "cover",
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "paddingHorizontal": 5,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "height": 25,
                  "width": 25,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "fullstar3",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": 25,
                      "width": 25,
                    },
                    Object {
                      "resizeMode": "cover",
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "paddingHorizontal": 5,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "height": 25,
                  "width": 25,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "fullstar4",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": 25,
                      "width": 25,
                    },
                    Object {
                      "resizeMode": "cover",
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "paddingHorizontal": 5,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "height": 25,
                  "width": 25,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "fullstar5",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": 25,
                      "width": 25,
                    },
                    Object {
                      "resizeMode": "cover",
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "flex": 2,
            "fontSize": 21,
            "textAlign": "right",
          }
        }
      >
        10
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 17,
              "height": 34,
              "width": 34,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#bdbdbd",
                "borderRadius": 17,
                "flex": 1,
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            <Image
              onLoadEnd={[Function]}
              source={
                Object {
                  "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/adhamdannaway/128.jpg",
                }
              }
              style={
                Object {
                  "flex": 1,
                  "height": null,
                  "width": null,
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            />
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#bdbdbd",
                    "flex": 1,
                    "height": null,
                    "justifyContent": "center",
                    "width": null,
                  }
                }
                testID="RNE__Image__placeholder"
              />
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "paddingLeft": 5,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
           
          
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text>
          查看详情
        </Text>
        <Image
          source={
            Object {
              "uri": "arrow_down",
            }
          }
          style={
            Object {
              "height": 15,
              "width": 15,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
